tools/development/gstPrototxt.py

```diff
@@ -0,0 +1,182 @@
+#!/usr/bin/env python3
+##
+# SPDX-License-Identifier: LGPL-2.1-only
+#
+# Prototxt <--> GStreamer pipeline converter
+# Copyright (c) 2020 Samsung Electronics
+#
+##
+# @file   gstPrototxt.py
+# @brief  Prototxt to/from GStreamer pipeline converter
+# @author MyungJoo Ham <myungjoo.ham@samsung.com>
+# @date   30 Jun 2020
+# @bug    No known bugs
+# @todo   WIP
+#
+
+## @todo Get python pbtxt parser
+
+## @todo property link up tensor-repo-*
+
+declaredNames = [] # list of names declared by user.
+generatedNameCounter = 0
+generatedNamePrefix = '__id'
+
+## @brief A class for a filter (object == filter instance)
+class Filter:
+    element = None    # element name (string)
+    name = None       # name property (string)
+    nameGenerated = False   # boolean
+    properties = [] # list of (key, value)
+    src = []        # list of (Filter, string(name of src), string(src-padname. maybe None))
+    sink = []        # list of (Filter, string(name of sink), string(sink-padname. maybe None))
+
+    def __init__(self, elementname, _name):
+        global generatedNameCounter
+        global generatedNamePrefix
+        global declaredNames
+
+        self.element = elementname
+        if _name is not None and len(_name) > 0:
+            name = _name
+            if name in declaredNames:
+                raise RuntimeError("Duplicated name in elements: " + str(name))
+            declaredNames.append(str(name))
+        else:
+            while ((generatedNamePrefix + str(generatedNameCounter)) not in declaredNames):
+                generatedNameCounter = generatedNameCounter + 1
+            name = generatedNamePrefix + str(generatedNameCounter)
+            nameGenerated = True
+
+## @brief A class for a pipeline (object == pipeline instance)
+#  @detail How to use (Gst pipeline)
+#            Phase 1: add all filters
+#            Phase 2: add all relations of !
+#          How to use (Pbtxt pipeline)
+#            Phase 1: add all filters
+#            Phase 2: add all src/sink relations
+class Pipeline:
+    filters = []    # list of Filter included in the pipeline
+    nameList = []
+
+    def __init__(self):
+        self.filters = []
+        self.srcs = []
+        self.sinks = []
+
+    ## @brief add a filter instnace
+    def addFilter(_filter):
+        self.filters.append(_filter)
+        self.nameList.append(_filter.name)
+
+    ## @brief Get filter with the given name
+    def __getFilterByName(name):
+        for f in self.filters:
```

Comment:
in python , its recommended to use `filter` than use a loop.
also, as nameList stores all the names of filters, checking directly in nameList would also suffice.

Reply:
Updated.